### DISCUSSION TITLE
Help with installation

### DISCUSSION THREAD

**Discussion by hamboneharr** [Lexpedite/blawx#390] [Q&A] [answered]
Greetings

I followed the install instructions.  The local website runs, but when I select "code editor" I do not see any blocks in the middle column.   I have tried both on Chrome and Edge browsers.  I do not have any restrictions on popups or javascript.  Something has to be amiss with the local installation as my browsers can display the public version at dev.blawx.com.  See screenshots attached.  

Does anyone have an FAQ on these types of problems?  Or any suggestions on how to approach it?  I wasn't sure where else to ask for help.

Other relevant info:

- I have also attached a printout of progress during the build.  Curiously the first line after I run $ ./update.sh
is "Error response from daemon: page not found" but the build continued OK.

- I have installed Docker Desktop v 4.14.1 on a Windows 10 and also a Windows 11 machine and followed instructions to update to "WSL2 backend". 

Really appreciate any tips at all as I am determined to try out blawx and currently I am stuck.

Regards

Hamish

[screenshots.docx](https://github.com/Lexpedite/blawx/files/10055016/screenshots.docx)
[install log file.txt](https://github.com/Lexpedite/blawx/files/10055017/install.log.file.txt)

**Comment by Gauntlet173** [comment]
Hi, Hamish!

Thanks for getting in touch. That's an odd problem.

update.sh looks for an existing instance to shut down that isn't there the first time you run it, so the error message is expected.

I wonder if you could open the developer tools inside your browser (CTRL+SHIFT+I), and tell me if there are any errors being thrown in the javascript console. It seems like it's not successfully injecting Blockly into the page, as opposed to not properly loading the blawx code into the blockly interface, so I suspect the javascript console might give us some clue as to what went wrong.

**Reply by hamboneharr** [reply]
Hi Jason

The developer tool shows 13 errors and going to the console (snippet attached) it has plenty of 404 (not found). 

Clicking on any of the 404 errors takes me to the scripts and I see there is nothing in there (further snippet attached).

I'm hoping these send us in a good direction.

![image](https://user-images.githubusercontent.com/104726768/203863853-c98b5697-29b9-4655-b1a6-2bd782ea7655.png)
![image](https://user-images.githubusercontent.com/104726768/203864497-3eff399b-0038-465d-a162-1478d2df896f.png)

Regards
Hamish

**Reply by Gauntlet173** [reply]
That's very helpful. The Dockerfile did not successfully grab the Blockly libraries and put them where they belong. Let me take another look at your install log and the Dockerfile, and I'll see if there is any indication as to why. Can you confirm you are using the current main branch (1.3.30-alpha is the version in settings.py)?

**Reply by Gauntlet173** [reply]
Yeah, the commands for moving the files where they are supposed to be weren't even attempted. I think we may have broken the Dockerfile when we made changes to speed it up. Sorry about that. This is what comes from not yet having a CI/CD workflow.

I will fix it and push a new version, but I'm traveling this weekend so it may not be until early next week. Thanks for your patience.

**Comment by hamboneharr** [comment]
Ah thanks very much for looking into and letting me know, I will stand by.  I procrastinated a while to try this, I can wait a bit longer!  I should also have mentioned I tried building with an earlier version 1.3.17-alpha but it had the same issue.

**Comment by Gauntlet173** [comment, marked as answer]
OK, I think I have figured out the problem. It seems that when Google's Blockly switched to version 9, they also changed what is included in the github repository, and where some of those things are kept. So there is now no way to install it that has both a) the compiled .js files, and b) the appengine file that we use for doing saving .blawx files.

The solution was to abandon the current installation method of cloning from git, and instead install using npm, and then grab a single file from the github repository that is not included in the npm package.

I have created a branch called "hamish" that has made the changes to the Dockerfile, and updated the HTML templates to agree with the new locations for some of the files in the package. Let me know if you have any problems, but I think it should work. If it works for you, I'll release it as v1.3.32-alpha.

**Comment by hamboneharr** [comment]
Awesome, I now see the blocks!  Thanks so much for sorting this out, now I can play :)
Regards
Hamish

**Reply by Gauntlet173** [reply]
I'll release the new version ASAP. Thanks very much for letting me know. Very excited to hear what you think.
